awsclient/s3.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package awsclient
+
+import (
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/pkg/errors"
+)
+
+const (
+	// appsS3BucketEnvVarName determines an environment variable.
+	// Variable saves address of apps S3 bucket name
+	AppsS3BucketEnvVarName = "MM_APPS_S3_BUCKET"
+
+	// defaultBucketName is the default s3 bucket name used to store app data.
+	DefaultBucketName = "mattermost-apps-bucket"
+)
+
+// GetS3 downloads files from S3.
+func (c *client) GetS3(bucket, item string) ([]byte, error) {
+	var buffer aws.WriteAtBuffer
+	_, err := c.s3Down.Download(&buffer, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(item),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to download file")
+	}
+	return buffer.Bytes(), nil
+}
+
+// S3FileUpload uploads file to a specific S3 bucket
+func (c *client) UploadS3(bucket, key string, body io.Reader) error {
+	if _, err := c.s3Uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+		Body:   body,
+	}); err != nil {
+		return errors.Wrap(err, "failed to upload file")
+	}
+	return nil
+}
+
+func (c *client) CreateS3Bucket(bucket string) error {
```

Comment:
nit: remove `Bucket` (here and below) or add it to the methods above.

Reply:
It creates the bucket, thus the distinction (others operate on entities in the bucket)